### PR TITLE
Nü Metrics: Mark I

### PR DESCRIPTION
**Pull request by eshurakov**
New Note Information card showing metrics (and later also interlink references)

@jleandroperez Please take a look when you have time. Thanks!

### Test (note)
1. Open a note

- [x] Check that there is a `(i)` button in the navigation bar

### Test (information card)

1. Open a note
2. Tap on the `(i)` button in the navigation bar

- [x] Information card is visible (also try when from is in landscape)
- [x] Information card shows Modified date, Created date, Words and Characters counts
- [x] Can be dismissed by tapping on `X`
- [x] Can be dismissed by tapping outside of the card
- [x] Can be dismissed by swiping the card down

### Release
Added in https://github.com/Automattic/simplenote-ios/pull/939/commits/5e35f83ac53cd1d9d7b49183350f9c0df95c75c7

> New Note Information interface #836

### CODE REVIEW

File: Simplenote/Classes/DateFormatter+Simplenote.swift [hunk 1]
```diff
@@ -7,12 +7,17 @@ extension DateFormatter {
...
     /// Shared DateFormatters
     ///
-    struct Simplenote {
-        static let listDateFormatter: DateFormatter = {
-            let formatter = DateFormatter()
-            formatter.dateStyle = .short
-            formatter.timeStyle = .none
-            return formatter
-        }()
-    }
```
Comment (jleandroperez) [inline Simplenote/Classes/DateFormatter+Simplenote.swift:17]: +1 That struct looked bad, thank you for dropping it!!

File: Simplenote/Classes/SPNoteEditorViewController+Extensions.swift [hunk 2]
```diff
@@ -35,6 +35,9 @@ extension SPNoteEditorViewController {
         checklistButton = UIBarButtonItem(image: .image(name: .checklist), style: .plain, target: self, action: #selector(insertChecklistAction(_:)))
         checklistButton.accessibilityLabel = NSLocalizedString("Inserts a new Checklist Item", comment: "Insert Checklist Button")
 
+        informationButton = UIBarButtonItem(image: .image(name: .info), style: .plain, target: self, action: #selector(noteInformationWasPressed(_:)))
+        informationButton.accessibilityLabel = NSLocalizedString("Information", comment: "Note Information Button (metrics + references)")
```
Comment (jleandroperez) [inline Simplenote/Classes/SPNoteEditorViewController+Extensions.swift:39]: I'm SO HAPPY to see this!!

(we had, up until last week, a 100% manually laid out NavigationBar titleView)!!

File: Simplenote/SPCardPresentationController.swift [hunk 3]
```diff
@@ -131,22 +136,26 @@ private extension SPCardPresentationController {
...
+            cardView.topAnchor.constraint(greaterThanOrEqualTo: containerView.centerYAnchor),
+        ]
+
         let heightConstraint = cardView.heightAnchor.constraint(equalToConstant: 0)
         heightConstraint.priority = .fittingSizeLevel
 
         NSLayoutConstraint.activate([
             cardView.bottomAnchor.constraint(equalTo: containerView.bottomAnchor),
-            cardView.topAnchor.constraint(greaterThanOrEqualTo: containerView.safeAreaLayoutGuide.topAnchor),
+            cardView.topAnchor.constraint(greaterThanOrEqualTo: containerView.safeAreaLayoutGuide.topAnchor, constant: 10),
```
Comment (jleandroperez) [inline Simplenote/SPCardPresentationController.swift]: Nipticky: Maybe move the `10` to a constant?

**Review by jleandroperez — APPROVED** [review]
Notes!

1. Info Sheet table rows can be pressed / highlighted
2. I've noticed the UITableView row separators look differently in iOS 12 (screenshots below)
3. It'd be neeeat if this UI would refresh automatically whenever the Note gets updated

Other than that, the code shines, end to end. Zero blockers, super good to address pendings elsewhere!

:shipit: sir!! Woooo!!

File: Simplenote/Classes/SPNoteEditorViewController+Extensions.swift (hunk 2 above)
Comment (eshurakov) [reply Simplenote/Classes/SPNoteEditorViewController+Extensions.swift:39]: Yes! This is definitely better 👍

File: Simplenote/SPCardPresentationController.swift (hunk 3 above)
Comment (eshurakov) [reply Simplenote/SPCardPresentationController.swift]: Good point, will do!

**Comment by eshurakov** [comment]
Thank you!
Good points, I will address them in the next PR.

**Pull request merged**